### ISSUE TITLE
Qt 5.14.2, libtorch1.7.0 cuda10.2，Error after using header file

### ISSUE THREAD

**Issue by kesoul** [pytorch/pytorch#48450]
## ❓ Questions and Help

Whether I use qmake or cmake to build a qt project that includes libtorch, it will be successful when I build it, but when I add the #include <torch/script.h> file to the .h file, click Run, The program will report an error. The error content of these two methods is exactly the same.

D:\libtorch\include\c10\core\DispatchKeySet.h:59: error: C3250: "<range>$L0": declaration in "constexpr" function body is not allowed
D:\libtorch\include\c10\core\DispatchKeySet.h:59: error: C3249: "constexpr" function statement or sub-expression is illegal
D:\libtorch\include\c10\core\DispatchKeySet.h:59: error: C3250: "<begin>$L0": declaration in "constexpr" function body is not allowed
D:\libtorch\include\c10\core\DispatchKeySet.h:59: error: C3250: "<end>$L0": declaration in "constexpr" function body is not allowed
D:\libtorch\include\c10\core\DispatchKeySet.h:59: error: C3250: "k": declaration in "constexpr" function body is not allowed
D:\libtorch\include\c10\core\DispatchKeySet.h:60: error: C3249: "constexpr" function statement or sub-expression is illegal
D:\libtorch\include\c10\util\ConstexprCrc.h:103: error: C3250: "i": declaration in "constexpr" function body is not allowed
D:\libtorch\include\c10\util\ConstexprCrc.h:103: error: C3249: "constexpr" function statement or sub-expression is illegal
D:\libtorch\include\c10\util\ConstexprCrc.h:104: error: C3249: "constexpr" function statement or sub-expression is illegal
D:\libtorch\include\c10\util\ConstexprCrc.h:105: error: C3249: "constexpr" function statement or sub-expression is illegal
D:\libtorch\include\c10\util\Optional.h:283: error: C2556: "c10::ScalarType &c10::optional<c10::ScalarType>::contained_val(void) const &": overloaded function and "Const c10::ScalarType &c10::optional<c10::ScalarType>::contained_val(void) const &" is only different in the return type

**Comment by skyline75489** [comment]
Seems like you're using an old version of Visual Studio?

**Comment by kesoul** [comment]
> Seems like you're using an old version of Visual Studio?

no, this error is the result of using qt only and qmake and cmake respectively. Today I tried it again. I used VS2019 and cmake to do it. It also reported an error and prompted torch-NOTFOUND, but I have configured the path in makelists.txt: set(Torch_DIR "D:/libtorch/share/cmake/Torch" ), but when I add the header file, I get an error: torch-NOTFOUND

**Comment by skyline75489** [comment]
The `torch-NOTFOUND` indicates that cmake failed to find a correct torch lib. This usually happens when you are mixing Debug & Release build. Say your configuration in the IDE is Debug, but you're using a Release version of Libtorch.
